### PR TITLE
feat: support openclip&mclip models + refactor model loader

### PR DESCRIPTION
**Pull request by shan-mx**

### CODE REVIEW

File: server/clip_server/model/mclip_model.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+class MultilingualCLIP(transformers.PreTrainedModel):
+    config_class = MCLIPConfig
+
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(config, *args, **kwargs)
+        self.transformer = transformers.AutoModel.from_pretrained(config.modelBase)
+        self.LinearTransformation = torch.nn.Linear(
+            in_features=config.transformerDimensions, out_features=config.numDims
```
Comment (numb3r3) [inline server/clip_server/model/mclip_model.py:52]: `config.transformerDimensions` seems like an integer, however, its default value is assigned with the string "xlm-roberta-large". Please double check.

File: server/clip_server/model/mclip_model.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        )
+
+    def encode_text(self, txt_tok):
```
Comment (numb3r3) [inline server/clip_server/model/mclip_model.py:55]:
```suggestion
    def encode_text(self, input_ids, attention_mask):
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]
Bassically, the `clip_model.py` is good. However, I just feel the `preprocessor.py` is an over-kill. I want to simplify this part in a new branch, and merge into this PR later.

File: server/clip_server/executors/clip_torch.py [hunk 3]
```diff
@@ -53,12 +56,13 @@ def __init__(
...
 
-        self._model, self._preprocess_tensor = clip.load(
-            name, device=self._device, jit=jit
-        )
+        self._model = CLIPModel(name, device=self._device, jit=jit, **kwargs)
+        self._tokenizer = Tokenizer(name)
 
-        self._pool = ThreadPool(processes=num_worker_preprocess)
+        image_size = get_image_size(self._model.visual_model_name)
+        self._image_transform = image_transform(image_size, is_train=False)
```
Comment (numb3r3) [inline server/clip_server/executors/clip_torch.py:65]: FYI, @shan-mx I changed my mind, I just push a new commit to use original image preprocessor.

File: server/clip_server/model/tokenization.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        if name in _MULTILINGUALCLIP_MODELS:
+            import transformers
+
+            self._tokenizer = transformers.AutoTokenizer.from_pretrained(name)
+        else:
+            from clip_server.model.simple_tokenizer import SimpleTokenizer
+
+            self._tokenizer = SimpleTokenizer()
+
+    def encode(
```
Comment (ZiniuYu) [inline server/clip_server/model/tokenization.py:18]: I suggest using a different name than `encode`, maybe `tokenize`, to distinguish from that in clip model

File: server/clip_server/model/pretrained_models.py [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+        patch_float(model.encode_image)
+        patch_float(model.encode_text)
+        model.float()
+
+    # ensure image_size attr available at consistent location for both jit and non-jit
+    model.visual.image_size = model.input_resolution.item()
+    return model
+
+
+def get_model_url_md5(model: str, pretrained: str):
```
Comment (numb3r3) [inline server/clip_server/model/pretrained_models.py:289]: why do we need this function?

File: server/clip_server/model/tokenization.py (hunk 4 above)
Comment (shan-mx) [reply server/clip_server/model/tokenization.py:18]: agree

File: server/clip_server/model/pretrained_models.py (hunk 5 above)
Comment (shan-mx) [reply server/clip_server/model/pretrained_models.py:289]: i dont understand this question. without this the download process cant work

File: server/clip_server/model/mclip_model.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    'M-CLIP/LABSE-Vit-L-14': ('ViT-L-14', 'openai'),
+}
+
+
+class MCLIPConfig(transformers.PretrainedConfig):
+    model_type = "M-CLIP"
+
+    def __init__(
+        self,
+        modelBase: str = 'xlm-roberta-large',
```
Comment (ZiniuYu) [inline server/clip_server/model/mclip_model.py:22]: lower case?

File: server/clip_server/model/tokenization.py (hunk 4 above)
Comment (numb3r3) [reply server/clip_server/model/tokenization.py]: done

**Comment by ZiniuYu** [comment]
86.61 wow new record

**Pull request merged**